client/engine/chainservice/eth_chain_helpers.go

```diff
@@ -0,0 +1,71 @@
+package chainservice
+
+import (
+	"errors"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	NitroAdjudicator "github.com/statechannels/go-nitro/client/engine/chainservice/adjudicator"
+)
+
+// getChainHolding reads on-chain holdings for a channel and an asset address given a transaction and an event generated by the transaction.
+func getChainHolding(na *NitroAdjudicator.NitroAdjudicator, tx *types.Transaction, event *NitroAdjudicator.NitroAdjudicatorAllocationUpdated) (common.Address, *big.Int, error) {
+	assetAddress, err := assetAddressForIndex(na, tx, event.AssetIndex)
+	if err != nil {
+		return assetAddress, &big.Int{}, err
+	}
+	amount, err := na.Holdings(&bind.CallOpts{BlockNumber: new(big.Int).SetUint64(event.Raw.BlockNumber)}, assetAddress, event.ChannelId)
+	if err != nil {
+		return assetAddress, amount, err
+	}
+	return assetAddress, amount, nil
+}
+
+// assetAddressForIndex uses the input parameters of a transaction to map an asset index to an asset address
+func assetAddressForIndex(na *NitroAdjudicator.NitroAdjudicator, tx *types.Transaction, index *big.Int) (common.Address, error) {
+	abi, err := NitroAdjudicator.NitroAdjudicatorMetaData.GetAbi()
+	if err != nil {
+		return common.Address{}, errors.New("unable to get ABI for a bound contract")
```

Comment:
This seems like a strange error. Can you explain it some more? Why not wrap or simply re-use the `err` form L29?

Reply:
Good point. Changed in 78dc149a.